File: src/wpsr/parsers/rdf/query/RDFNode.java
package wpsr.parsers.rdf.query;

public class RDFNode implements IRDFNode {

    private com.hp.hpl.jena.rdf.model.RDFNode node = null;

    public RDFNode(com.hp.hpl.jena.rdf.model.RDFNode node) {
	this.node = node;
    }

    public RDFNode(wpsr.parsers.rdf.query.IRDFNode node) {
	this.node = (com.hp.hpl.jena.rdf.model.RDFNode) node.extractRawObject();
    }

    @Override
    public Object extractRawObject() {
	return node;
    }

    @Override
    public IRDFNode as(Class<?> arg0) {
	return new RDFNode(node.as(arg0));
    }

    @Override
    public boolean canAs(Class<?> arg0) {
	return node.canAs(arg0);
    }

    @Override
    public boolean isAnon() {
	return node.isAnon();
    }

    @Override
    public boolean isLiteral() {
	return node.isLiteral();
    }

    @Override
    public boolean isResource() {
	return node.isResource();
    }

    @Override
    public boolean isURIResource() {
	return node.isURIResource();
    }

    @Override
    public String toString() {
	return node.toString();
    }
}

File: src/wpsr/parsers/rdf/query/RDFNode.ts
///<reference path="./IRDFNode.d.ts" />

module wpsr.parsers.rdf.query {
  class RDFNode implements IRDFNode {

    // TODO: get RDFNode from jena
    private com.hp.hpl.jena.rdf.model.RDFNode node = null;

    constructor(com.hp.hpl.jena.rdf.model.RDFNode node) {
      this.node = node;
    }

    constructor(wpsr.parsers.rdf.query.IRDFNode node) {
      this.node = (com.hp.hpl.jena.rdf.model.RDFNode) node.extractRawObject();
    }

    public extractRawObject(): Object {
      return node;
    }

    public transform(obj: Object): IRDFNode {
      return new RDFNode(node.as(obj));
    }

    public canTransform(obj: Object): bool {
      return node.canAs(obj);
    }

    public isAnon(): bool {
      return node.isAnon();
    }

    public isLiteral(): bool {
      return node.isLiteral();
    }

    public isResource(): bool {
      return node.isResource();
    }

    public isURIResource(): bool {
      return node.isURIResource();
    }

    public toString(): String {
      return node.toString();
    }
  }
}
